### PR TITLE
Client API refactoring - limiting errors to crate-level error types

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

**Comment by gavofyork** [comment]
this seems to have added code and faff with no obvious reasons for it.

**Comment by NikVolf** [comment]
@gavofyork 
nope, just removed from one place, inserted in the place where errors can be serialized and added mapping

needed for rpc api

File: ethcore/src/types/transaction_import.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	Current,
+	/// Transaction was imported to future queue.
+	Future
+}
+
+binary_fixed_size!(TransactionImportResult);
+
+/// Api-level error for transaction import
+#[derive(Debug, Clone, Binary)]
+pub enum TransactionImportError {
```
Comment (debris) [inline ethcore/src/types/transaction_import.rs]: Is this type necessary? Can't we just use `TransactionError`?

File: ethcore/src/types/transaction_import.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	Transaction(TransactionError),
+	/// Other error
+	Other(String),
+}
+
+impl From<Error> for TransactionImportError {
+	fn from(e: Error) -> Self {
+		match e {
+			Error::Transaction(transaction_error) => TransactionImportError::Transaction(transaction_error),
+			_ => TransactionImportError::Other(format!("other block import error: {:?}", e)),
```
Comment (debris) [inline ethcore/src/types/transaction_import.rs]: imo, we should never do this type of conversion, cause it makes `TransactionImportError` more generic then `Error`

File: ethcore/src/client/client.rs [hunk 3]
```diff
@@ -778,12 +780,16 @@ impl BlockChainClient for Client {
...
 		let fetch_account = |a: &Address| AccountDetails {
 			nonce: self.latest_nonce(a),
 			balance: self.latest_balance(a),
 		};
-		self.miner.import_transactions(self, transactions, fetch_account)
+
+		self.miner.import_transactions(self, transactions, &fetch_account)
+			.into_iter()
+			.map(|res| res.map_err(|e| e.into()))
+			.collect()
```
Comment (debris) [inline ethcore/src/client/client.rs]: `self.miner.import_transactions` should return `Result<_, TransactionImportError>` so this code is redundant

**Comment by debris** [comment]
I see that this pr is already merged. I generally like the changes that it introduced, but it needs some polishing

**Pull request merged**